### PR TITLE
Support parquet page filtering for decimal128 columns

### PR DESCRIPTION
**Pull request by Ted-Jiang**
Signed-off-by: yangjiang <yangjiang@ebay.com>

# Which issue does this PR close?

Closes #.

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/src/physical_plan/file_format/parquet/page_filter.rs [hunk 1]
```diff
@@ -394,16 +407,50 @@ macro_rules! get_min_max_values_for_page_index {
         match $self.col_page_indexes {
             Index::NONE => None,
             Index::INT32(index) => {
-                let vec = &index.indexes;
-                Some(Arc::new(Int32Array::from_iter(
-                    vec.iter().map(|x| x.$func().cloned()),
-                )))
+                match $self.target_type {
+                    // int32 to decimal with the precision and scale
```
Comment (Ted-Jiang) [inline datafusion/core/src/physical_plan/file_format/parquet/page_filter.rs:411]: Here only support `decimal128`  align with row group pruning.

File: datafusion/core/src/physical_plan/file_format/parquet/page_filter.rs [hunk 2]
```diff
@@ -394,16 +407,50 @@ macro_rules! get_min_max_values_for_page_index {
...
+                    Some(DataType::Decimal128(precision, scale)) => {
+                        let vec = &index.indexes;
+                        if let Ok(arr) = Decimal128Array::from_iter_values(
+                            vec.iter().map(|x| *x.$func().unwrap() as i128),
```
Comment (Ted-Jiang) [inline datafusion/core/src/physical_plan/file_format/parquet/page_filter.rs:415]: https://github.com/apache/arrow-rs/blob/d88ed6a4eb72eb2f87544f2811c23fd55dc706be/parquet/src/file/statistics.rs#L434

i think its ok `unwrap` here, same as  row group pruning

**Review by Ted-Jiang — COMMENTED** [review]
@alamb Could you take a look ? 😄

**Comment by Ted-Jiang** [comment]
Still need fix the Conflicting with up/master 😩 , I try to rebase but lost your commit.

**Comment by Ted-Jiang** [comment]
CI now fail
```
use of deprecated associated function `chrono::NaiveDate::from_ymd`: use `from_ymd_opt()` instead
````
I think this fix by up/master, I prefer merge this `string` and `decimal128` to  master,  we can do deal with nulls on follow on PRs

**Comment by alamb** [comment]
I am going to merge this PR into https://github.com/apache/arrow-datafusion/pull/4132 and we can keep working there (feel free to push commits directly to my branch)

**Pull request merged**